pkg/tfshim/sdk-v2/provider_test.go

```diff
@@ -104,3 +107,209 @@ func TestProvider1UpgradeResourceState(t *testing.T) {
 		})
 	}
 }
+
+func TestProviderDetailedSchemaDump(t *testing.T) {
+	prov := NewProvider(&schema.Provider{
+		ResourcesMap: map[string]*schema.Resource{
+			"test_resource": {
+				Schema: map[string]*schema.Schema{
```

Comment:
We should probably have a test for blocks (with nesting levels) and for nested types.